script.js

```diff
@@ -1,4 +1,696 @@
+      allPlayers[dealIndex].player.currentBet * 2;
+    return `You have gotten a blackjack and won!<br><br>${playerHandText} <br><br> The computer did not get a blackjack. Your current bankroll is now ${allPlayers[dealIndex].player.bankroll}`;
+  } else {
+    return `${playerHandText}`;
+  }
+};
+
+var playerTurnFunc = function (input) {
+  var myOutputValue = "";
+  if (mode == "playerDecisionTurn" || mode == "splitDecision") {
+    if (allPlayers[cardDisplayCounter].player.blackjackCheck == true) {
+      mode = "playerEndTurn";
+      return `Since you have gotten a blackjack, please pass it on to the next player.`;
+    } else {
+      playerTotalValue = 0;
+      if (allPlayers[cardDisplayCounter].player.wasSplit == false) {
+        playerHandText = `Your hand: <br>${allPlayers[cardDisplayCounter].player.playerHand[0].name} of ${allPlayers[cardDisplayCounter].player.playerHand[0].emojiSuit} <br> ${allPlayers[cardDisplayCounter].player.playerHand[1].name} of ${allPlayers[cardDisplayCounter].player.playerHand[1].emojiSuit}`;
+      } else {
+        playerHandText = `Your hand: <br>${allPlayers[cardDisplayCounter].player.playerHand[0].name} of ${allPlayers[cardDisplayCounter].player.playerHand[0].emojiSuit}`;
+      }
+      if (allPlayers[cardDisplayCounter].player.wasSplit == false) {
+        if (
+          mode == "playerDecisionTurn" &&
+          allPlayers[cardDisplayCounter].player.playerHand[0].rank ==
+            allPlayers[cardDisplayCounter].player.playerHand[1].rank
+        ) {
+          mode = "splitDecision";
+          return `${playerHandText}. You can choose to split these hands into 2. Do you want to? (yes/no)`;
+        }
+        if (mode == "splitDecision") {
+          if (input == "yes") {
+            noOfPlayers += 1;
+            allPlayers[cardDisplayCounter].player.hasSplit = true;
+            splitPlayer = {
+              player: {
+                username: allPlayers[cardDisplayCounter].player.username,
+                playerHand: [
+                  allPlayers[cardDisplayCounter].player.playerHand[1],
+                ],
+                handTotal: 0,
+                handText: "",
+                wasSplit: true,
+              },
+            };
+            allPlayers.splice([cardDisplayCounter + 1], 0, splitPlayer);
+            allPlayers[cardDisplayCounter].player.playerHand.pop(1);
+            playerHandText = `You have split your hand into 2. They will be taken as separate hands. This hand:  <br>${allPlayers[cardDisplayCounter].player.playerHand[0].name} of ${allPlayers[cardDisplayCounter].player.playerHand[0].emojiSuit}`;
+          } else if (input == "no") {
+            playerHandText += `<br> Your hand remains as per normal.`;
+          } else {
+            return `Please only input either yes or no!`;
+          }
+          mode = "playerturn";
+          return playerHandText;
+        }
+      }
+      mode = "playerturn";
+      return playerHandText;
```

Comment:
this is quite an extensive nesting of if and else ifs too. Generally, I would recommend that you can take our more logic intensive ifs and place it as its own helper function.